salsah/src/public/js/jquery.searchbox.js

```diff
@@ -206,7 +208,12 @@
 							left: pos.left,
 							top: pos.top
 						}).show();
-						$this.val('');
+						// if the input already has a value and it is valid, back it up
```

Comment:
I did so. ("stringified" `true`/`false` values I set `isValid`) in 5beebbd8e6c83f1d0d70af56eafdb5f32a332a17

Thanks for the "javascript, the good parts" reference!